data/projects/projectsRouter.js

```diff
@@ -0,0 +1,104 @@
+const express = require("express");
+const Projects = require("../helpers/projectModel");
+const Actions = require("../helpers/actionModel");
+const router = express.Router();
+
+router.get("/", async (req, res) => {
+  try {
+    const projects = await Projects.get();
+    res.status(200).json(projects);
+  } catch (error) {
+    res.status(500).json(error);
+  }
+});
+
+router.get("/:id", validateId, async (req, res) => {
+  try {
+    project = req.project;
+    res.status(200).json(project);
+  } catch (error) {
+    res.status(500).json(error);
+  }
+});
+
+router.get("/:id/actions", async (req, res) => {
+  try {
+    const id = req.params.id;
+    const actions = await Projects.getProjectActions(id);
+
+    res.status(200).json(actions);
+  } catch (error) {
+    res.status(500).json(error);
+  }
+});
+
+router.post("/", async (req, res) => {
+  try {
+    const projectAddition = {
+      name: req.body.name,
+      description: req.body.description
+    };
+    const project = await Projects.insert(projectAddition);
+
+    res.status(201).json(project);
+  } catch (error) {
+    res.status(500).json(error);
+  }
+});
+
+router.post("/:id/actions", async (req, res) => {
+  try {
+    const actionAddition = {
+      description: req.body.description,
+      notes: req.body.notes,
+      project_id: req.params.id
+    };
+    const action = await Actions.insert(actionAddition);
+
+    res.status(201).json(action);
+  } catch (error) {
+    res.status(500).json(error);
+  }
+});
+
+router.delete("/:id", async (req, res) => {
```

Comment:
You could use your validate id middleware on the put and delete endpoints. Currently you don't have a condition to send the client if the id doesn't exist in the database.